### PR TITLE
docs: added feature doc & tutorial for native frame navigation

### LINKED ISSUE #3660: Document native Frame navigation with Uno
## What would you like clarification on:
Outline the steps and best practices for adding page navigation in the shared project. 

## Concern?

- [ ] Usage in industry
- [ ] Clarification of capabilities
- [ ] Getting started with Uno
- [X ] Developing with Uno
- [ ] Contributing to the Uno project
- [ ] Publishing your application
- [ ] Support
- [ ] Other (please specify):

## For which Platform:

- [X ] iOS
- [X ] Android
- [X ] WebAssembly
- [X ] macOS
- [X ] Windows

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable):
resolves #3660

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## What is the new behavior?

added guide on frame navigation & docs on native frame navigation

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
Related sample code is in another PR: unoplatform/Uno.Samples#44

### CODE REVIEW

File: doc/articles/guides/native-frame-nav-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        ```
+    - `MainPage.xaml.cs`, `BlankPage1.xaml.cs`:
+        ```cs
+        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage1)); // in MainPage
+        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage2)); // in BlankPage1
+        ```
+5. (Optionally) Add an `AppBarButton` to the `CommandBar` for back navigation in the last two pages for Skia heads:
+    - `BlankPage1.xaml`, `BlankPage2.xaml`:
+        > note: Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more informations on this feature, check out: <todo: link>
+        ```xml
```
Comment (ajpinedam) [inline doc/articles/guides/native-frame-nav-tutorial.md:45]: Seems it's missing the link

File: doc/articles/features/native-frame-nav.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Native Frame Navigation
+
+## Introduction
+Since Uno.UI is written under the UWP api contract, most of the how-tos are sharable. In this article, we will discuss the particularities of the native `Frame` navigation in detail.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:4]:
```suggestion
This article discusses the particularities of the native `Frame` navigation in detail.
```

File: doc/articles/features/native-frame-nav.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+It is recommended to be familiar with the concept of navigation in UWP, here are some good starting points:
+- [Navigation design basics for Windows apps](https://docs.microsoft.com/en-us/windows/uwp/design/basics/navigation-basics)
+- [`Windows.UI.Xaml.Controls.Frame`](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Controls.Frame)
+
+## Native Frame Navigation
+On Android and iOS, there is the option to enable the native frame navigation which replaces below controls' UWP style & template by their native counterpart:
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:11]:
```suggestion
On Android and iOS, there is the option to enable native frame navigation. Enabling this feature replaces the default styles and templates of the controls below:
```

File: doc/articles/features/native-frame-nav.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+- `Windows.UI.Xaml.Controls.Frame`<superscript>*</superscript>
+- `Windows.UI.Xaml.Controls.CommandBar`
+- `Windows.UI.Xaml.Controls.AppBarButton`
+
+This feature can be enabled by calling the method below typically in the constructor of `App` class:
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:16]:
```suggestion
This feature can be enabled by calling the method below, typically in the constructor of your `App` class in `App.xaml.cs`:
```

File: doc/articles/features/native-frame-nav.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+```cs
+#if __IOS__ || __ANDROID__
+    Uno.UI.FeatureConfiguration.Style.ConfigureNativeFrameNavigation();
+#endif
+```
+> note: It is recommended to enable this feature in order to provide a more native user experience.
+
+### CommandBar and AppBarButton
+In essence, by enabling this, a custom renderer will be used as the control template. This renderer injects the native control in lieu and acts as a mediator, that maps dependency properties (colors, content/icon, ...) and connects events (clicked), between the UWP controls and the native ones:
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:25]:
```suggestion
In essence, by enabling this, a custom renderer will be used as the control template. This renderer injects the native control into the app, and maps dependency properties (colors, content/icon, ...) and events (clicked), between the UWP controls and the native ones:
```

File: doc/articles/features/native-frame-nav.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+UWP Control|Native Android Control|Native iOS Control
+-|-|-
+`CommandBar`|`android.widget.Toolbar`|`UIKit.UINavigationBar`
+`AppBarButton`|`android.view.MenuItem`|`UIKit.UIBarButtonItem`
+
+As such, you can achieve the feel and look of native control, including feature like native swipe to go back on iOS, while remaining in the familiar territory of UWP. However, you are no longer able to customize the control template.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:32]:
```suggestion
As such, you can have the look and feel of the native control, including features like native swipe to go back on iOS, while programming against the UWP API. However, you are no longer able to customize the control template.
```

File: doc/articles/features/native-frame-nav.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+### Frame
+The `Frame` also uses a custom presenter for its control template.
+On iOS, this presenter creates an `UINavigationController`, and keeps its states in sync with its `Frame`. It also creates a `UIViewController` for each `Page` that gets navigated.
+On iOS and on Android, it also manages the state of back button on the `CommandBar`, if it is presents on a `Page`.
+
+## Platform Specifics
+- android: Tapping the CommandBar's back button, system back button, or performing the sweep gesture will trigger `SystemNavigationService.BackRequested`. It's the responsibility of the application's navigation controller to eventually call `Frame.GoBack()`.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:40]:
```suggestion
- Android: Tapping the CommandBar's back button, system back button, or performing the sweep gesture will trigger `SystemNavigationService.BackRequested`. It's the responsibility of the application's navigation controller to eventually call `Frame.GoBack()`.
```

File: doc/articles/features/native-frame-nav.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+- ios: Tapping the back button automatically triggers a back navigation on the native `UINavigationController` (which is part of the native `Frame` implementation). The `Frame` will raise the `Navigated` event, and its state will automatically be updated to reflect the navigation. The navigation can't be intercepted or canceled.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:41]:
```suggestion
- iOS: Tapping the back button automatically triggers a back navigation on the native `UINavigationController` (which is part of the native `Frame` implementation). The `Frame` will raise the `Navigated` event, and its state will automatically be updated to reflect the navigation. The navigation can't be intercepted or canceled.
```

File: doc/articles/features/native-frame-nav.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+- any uno platforms: `Uno.UI.FeatureConfiguration.Page.IsPoolingEnabled` can be set to enable reuse of `Page` instances. Enabling this allows the pages in the forward stack to be recycled when stack is modified, which can improve performance of `Frame` navigation.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:42]:
```suggestion
- all non-Windows platforms: `Uno.UI.FeatureConfiguration.Page.IsPoolingEnabled` can be set to enable reuse of `Page` instances. Enabling this allows the pages in the forward stack to be recycled when stack is modified, which can improve performance of `Frame` navigation.
```

File: doc/articles/features/native-frame-nav.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+- android: On mobile, the pages in the back stack are still preserved in the visual tree for performance reason. This feature can be disabled on Android through `Uno.UI.FeatureConfiguration.NativeFramePresenter.AndroidUnloadInactivePages`.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:43]:
```suggestion
- Android: On mobile, the pages in the back stack are still preserved in the visual tree for performance reasons. This feature can be disabled on Android by setting `Uno.UI.FeatureConfiguration.NativeFramePresenter.AndroidUnloadInactivePages = true`.
```

File: doc/articles/features/native-frame-nav.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+- wasm: `SystemNavigationManager.AppViewBackButtonVisibility` needs to be set to `Visible` to intercept the browser back navigation. The intercepted event is forwarded to `SystemNavigationService.BackRequested`. It's the responsibility of the application's navigation controller to eventually call `Frame.GoBack()`.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:44]:
```suggestion
- WASM: `SystemNavigationManager.AppViewBackButtonVisibility` needs to be set to `Visible` to intercept the browser back navigation. The intercepted event is forwarded to `SystemNavigationService.BackRequested`. It's the responsibility of the application's navigation controller to eventually call `Frame.GoBack()`.
```

File: doc/articles/guides/native-frame-nav-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+## Example
+The complete sample code can be found here: [NativeFrameNav](https://github.com/unoplatform/Uno.Samples/tree/master/UI/NativeFrameNav)
+
+## Step-by-step instructions
+1. Create a new Uno Platform application, following the instructions [here](..\get-started.md).
+2. Add two more pages (`BlankPage1` and `BlankPage2`) to the `Your_Project_Name.Shared` project
+
+    Right click on the shared project -> `Add` -> `New Item...` -> `Blank Page`
+    Repeat once
+3. Modify of the content of each page to:
```
Comment (davidjohnoliver) [inline doc/articles/guides/native-frame-nav-tutorial.md:12]:
```suggestion
3. Modify the content of each page to:
```

File: doc/articles/features/native-frame-nav.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+- `Windows.UI.Xaml.Controls.CommandBar`
+- `Windows.UI.Xaml.Controls.AppBarButton`
+
+This feature can be enabled by calling the method below typically in the constructor of `App` class:
+```cs
+#if __IOS__ || __ANDROID__
+    Uno.UI.FeatureConfiguration.Style.ConfigureNativeFrameNavigation();
+#endif
+```
+> note: It is recommended to enable this feature in order to provide a more native user experience.
```
Comment (davidjohnoliver) [inline doc/articles/features/native-frame-nav.md:22]:
```suggestion
> [!NOTE]
> It is recommended to enable this feature in order to provide a more native user experience.
```

(This is docfx syntax to add a note block: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#note-warningtipimportant )

File: doc/articles/guides/native-frame-nav-tutorial.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+        </StackPanel>
+        ```
+    - `MainPage.xaml.cs`, `BlankPage1.xaml.cs`:
+        ```cs
+        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage1)); // in MainPage
+        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage2)); // in BlankPage1
+        ```
+5. (Optionally) Add an `AppBarButton` to the `CommandBar` for back navigation in the last two pages for Skia heads:
+    - `BlankPage1.xaml`, `BlankPage2.xaml`:
+        > note: Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more information on this feature, check out: [platform specific xaml](doc\articles\platform-specific-xaml)
```
Comment (davidjohnoliver) [inline doc/articles/guides/native-frame-nav-tutorial.md:44]:
```suggestion
        > [!NOTE]
        > Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more information on this feature, check out: [platform specific XAML](../platform-specific-xaml.md)
```

File: doc/articles/guides/native-frame-nav-tutorial.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+                e.Handled = true;
+            }
+        };
+    #endif
+    }
+
+## Additional Resources
+- [Uno-specific documentation](../controls/CommandBar) on `CommandBar` and `AppBarButton`
+- [Native Frame navigation](../features/native-frame-nav)
```
Comment (davidjohnoliver) [inline doc/articles/guides/native-frame-nav-tutorial.md:120]:
```suggestion
- [Native Frame navigation](../features/native-frame-nav.md)
```

File: doc/articles/toc.yml [hunk 16]
```diff
@@ -256,6 +258,8 @@
               href: features\ElevatedView.md
             - name: Other features
               href: Uno.UI.Toolkit.md
+        - name: Native Frame Navigation
```
Comment (davidjohnoliver) [inline doc/articles/toc.yml:261]: put this under 'WinUI features', since Frame navigation is a WinUI feature that has some Uno-specific functionality.

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**